### PR TITLE
Succesful quest

### PR DESCRIPTION
**Pull request by maksimpro001**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, dis: Distance | int | float) -> Distance:
+        if isinstance(dis, Distance):
+            return Distance(self.km + dis.km)
+        else:
+            return Distance(self.km + dis)
```
Comment (EdAlekseiev) [inline app/main.py:18]:
```suggestion
        return Distance(self.km + dis)
```

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+        return Distance(round(self.km * dis, 3))
+
+    def __truediv__(self, dis: int | float) -> Distance:
+        return Distance(round(self.km / dis, 2))
+
+    def __lt__(self, dis: Distance | float | int) -> bool:
+        if isinstance(dis, Distance):
+            return self.km < dis.km
+        else:
+            return self.km < dis
```
Comment (EdAlekseiev) [inline app/main.py:37]: The same comment as above

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __gt__(self, dis: Distance | float | int) -> bool:
+        if isinstance(dis, Distance):
+            return self.km > dis.km
+        else:
+            return self.km > dis
```
Comment (EdAlekseiev) [inline app/main.py:43]: The same comment as above

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __eq__(self, dis: Distance | float | int) -> bool:
+        if isinstance(dis, Distance):
+            return self.km == dis.km
+        else:
+            return self.km == dis
```
Comment (EdAlekseiev) [inline app/main.py:49]: The same comment as above

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __le__(self, dis: Distance | float | int) -> bool:
+        if isinstance(dis, Distance):
+            return self.km <= dis.km
+        else:
+            return self.km <= dis
```
Comment (EdAlekseiev) [inline app/main.py:55]: The same comment as above

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __ge__(self, dis: Distance | float | int) -> bool:
+        if isinstance(dis, Distance):
+            return self.km >= dis.km
+        else:
+            return self.km >= dis
```
Comment (EdAlekseiev) [inline app/main.py:61]: The same comment as above

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,55 @@
...
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, dis: Distance | int | float) -> Distance:
+        if isinstance(dis, Distance):
+            return Distance(self.km + dis.km)
+        return Distance(self.km + dis)
+
+    def __iadd__(self, dis: Distance | int | float) -> None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:19]: It return Distance, not None

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by YukihiroSM — APPROVED** [review]
Good job!

**Pull request closed without merging**